pkg/operator/controller/canary/controller.go

```diff
@@ -313,11 +335,28 @@ func (r *reconciler) startCanaryRoutePolling(stop <-chan struct{}) error {
 		}
 		successiveFail = 0
 		errors = []timestampedError{}
-		// Only increment checkCount if periodic canary route
-		// endpoint rotation is enabled to prevent unbounded
-		// integer growth.
+
+		// Check if canary route rotations are enabled every iteration.
+		rotationEnabled := r.isCanaryRouteRotationEnabled()
+		// Increment checkCount and periodically rotate the canary route endpoint if canary route rotation is enabled.
 		if rotationEnabled {
 			checkCount++
+			if checkCount > canaryCheckCycleCount {
```

Comment:
If you set `canaryCheckFrequency` to 8s and `canaryCheckCycleCount=` then you have to wait 16s.

```suggestion
                        if checkCount >= canaryCheckCycleCount {
```